src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -85,10 +85,38 @@
 (def cli-description "Main PuppetDB daemon")
 (def database-metrics-registry (get-in metrics/metrics-registries [:database :registry]))
 
-;; ## Wiring
-;;
-;; The following functions setup interaction between the main
-;; PuppetDB components.
+(def clean-options
+  #{"expire_nodes" "purge_nodes" "purge_reports" "package_gc" "other"})
+
+(def purge-nodes-opts-schema {:batch_limit s/Int})
+
+(def clean-request-schema
+  ;; i.e. a possibly empty collection of elements, each of which must
+  ;; either be a string from clean-options or a purge_nodes command
+  ;; with an options map.
+  [(s/if string?
```

Comment:
Is there a schema `or`?  I thought you were supposed to use `conditional`, `cond-pre`, or, `if` now.